druid/examples/image_encapsulate.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This example shows how to draw an png image.
+//!
+//! Requires the non-default "image" feature to be enabled:
+//! `cargo run --example image_encapsulate --features "images"`
+//!
+
+#[cfg(not(feature = "image"))]
+fn main() {
+    eprintln!("This examples requires the \"image\" feature to be enabled:");
+    eprintln!("cargo run --example image_encapsulate --features \"image\"");
+}
+
+#[cfg(feature = "image")]
+fn main() {
+    use druid::{
+        widget::{Flex, Image, ImageData, Slider, WidgetExt},
+        AppLauncher, BoxConstraints, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+        PaintCtx, Size, UpdateCtx, Widget, WindowDesc,
+    };
+
+    pub struct ImageSwitcher {
+        image_index: u32,
+        images: Vec<Box<dyn Widget<u32>>>,
+    }
+
+    impl ImageSwitcher {
+        pub fn new(newimages: Vec<Box<dyn Widget<u32>>>) -> Self {
+            Self {
+                image_index: 0,
+                images: newimages,
+            }
+        }
+    }
+
+    impl Widget<f64> for ImageSwitcher {
```

Comment:
semantically, it feels like the data type for this widget should also be `usize`, assuming it is supposed to represent an index. If you want to modify that data with a slider, you could use a lens to map the float to an int.